product_unique_serial/model/stock.py

```diff
@@ -54,78 +54,34 @@ def _get_last_location_id(self):
 class StockQuant(models.Model):
     _inherit = 'stock.quant'
 
-    @api.model
-    def _quant_create(self, qty, move, lot_id=False, owner_id=False,
-                      src_package_id=False, dest_package_id=False,
-                      force_location_from=False, force_location_to=False):
-        # In case of 'Unique Lot' check if the product does not exist somewhere
-        # internally already
-        if lot_id and move.product_id.lot_unique_ok:
-            # Extract from Odoo v9
-            if qty != 1.0:
-                raise exceptions.Warning(_('You should only receive by '
-                                           'the piece with the same serial '
-                                           'number'))
-            # Customize domain
-            domain_quants = [
-                ('product_id', '=', move.product_id.id),
-                ('lot_id', '=', lot_id),
-                ('qty', '>', 0.0)
-            ]
-            # Check product tracking field to get location usage for domain
-            if move.product_id.track_all:
-                domain_quants += [('location_id.usage', '!=', 'inventory')]
-            else:
-                usages = []
-                if move.product_id.track_incoming:
-                    usages += ['internal']
-                if move.product_id.track_outgoing:
-                    usages += ['customer', 'transit']
-                # mrp module should be installed to use track production field
-                if hasattr(move.product_id, "track_production") and \
-                        move.product_id.track_production:
-                    usages += ['production']
-                if usages:
-                    domain_quants += [
-                        ('location_id.usage', 'in', tuple(usages))]
-            # check if exist other similar quant
-            if self.search(domain_quants):
-                lot_name = self.env['stock.production.lot'].browse(lot_id).name
-                raise exceptions.Warning(_('The serial number %s can only '
-                                           'belong to a single product in '
-                                           'stock') % lot_name)
-        return super(StockQuant, self)._quant_create(
-            qty, move, lot_id, owner_id, src_package_id,
-            dest_package_id, force_location_from, force_location_to)
-
     @api.multi
     @api.constrains('product_id', 'lot_id', 'qty')
     def _check_inicity_lot_product(self):
```

Comment:
> s/inicity/uniqueness/g

> s/_check_inicity_lot_product/_check_uniqueness_lot/g

@nhomar 
A long time ago that was taken care of

check https://github.com/Vauxoo/stock-logistics-workflow/pull/13

Regards